message_hub/src/main/java/ordina/youbionic/infrastructure/MessagingConfig.java

```diff
@@ -0,0 +1,56 @@
+package ordina.youbionic.infrastructure;
+
+import org.springframework.amqp.core.*;
+import org.springframework.amqp.rabbit.connection.CachingConnectionFactory;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.amqp.rabbit.connection.ConnectionFactory;
+import org.springframework.amqp.support.converter.Jackson2JsonMessageConverter;
+import org.springframework.amqp.support.converter.MessageConverter;
+
+
+// Currently uncertain whether I need this class or not. Will be doing some experiments soon
+//TODO: Figure out if this config is (still) necessary or not.
+@Configuration
+public class
+MessagingConfig {
```

Comment:
2 points:
1. Which architecture are you going to use as in the folder structure but also how the code will be structured, so you got two different architecture types to chose from, a software architecture Pattern and a design pattern for withing the code. So the first one is for high level meaning, folder structuring and how your application is going to react and The other one is more low based which will deside on how to solve certain issues on components ord a specific problem to solve eq publish a message to rabbitmq

2. The configuration of most rabbitmq stuff can be done in the application.properties file -> https://docs.spring.io/spring-boot/docs/current/reference/html/application-properties.html#application-properties.integration.spring.rabbitmq.port search for rabbitmq properties and you will find enough